### ISSUE TITLE
thread 'main' has overflowed its stack

### ISSUE THREAD

**Issue by ddavness** [JohnnyMorganz/wally-package-types#9]
I've tested this on Windows with the **release** version downloaded/managed via aftman.
This error only seems to happen on certain dependencies.

```toml
# aftman.toml
[tools]
rojo = "rojo-rbx/rojo@7.4.1"
wally = "UpliftGames/wally@0.3.2"
wally-types = "JohnnyMorganz/wally-package-types@1.3.1"
```

```toml
# wally.toml
[package]
name = "some/reponame"
version = "0.1.0"
registry = "https://github.com/UpliftGames/wally-index"
realm = "shared"
private = true

[server-dependencies]
ProfileService = "tazmondo/profileservice@>=1.0.6"
```

> **Note:** This is probably dependency-dependant. This dependency in particular repros the issue.

Minimal `default.project.json` that can repro:

```json
{
  "name": "test",
  "tree": {
    "$className": "DataModel",
    "ServerScriptService": {
      "Server": {
        "$path": "ServerPackages"
      }
    },
    "Workspace": {
      "$properties": {
        "FilteringEnabled": true
      }
    }
  }
}
```

Running the following commands results in a stack overflow:

```
PS C:\Users\user\Documents\git\test> wally install
[INFO ] Updating package index https://github.com/UpliftGames/wally-index...
   Resolved 1 dependencies
  Generated lockfile
    Cleaned package destination
 Downloaded tazmondo/profileservice@1.0.6
[INFO ] Downloaded 1 packages!
PS C:\Users\user\Documents\git\test> rojo sourcemap -o sourcemap.json
Created sourcemap at sourcemap.json
PS C:\Users\user\Documents\git\test> wally-types --sourcemap .\sourcemap.json .\ServerPackages\
info: Found link file '.\ServerPackages\ProfileService.lua'
info: Require expression converted to path: 'script/Parent/_Index/tazmondo_profileservice@1.0.6/profileservice'
info: Link require points to profileservice [ModuleScript] @ '\\?\C:\Users\user\Documents\git\test\ServerPackages\_Index\tazmondo_profileservice@1.0.6\profileservice\ProfileService.lua'

thread 'main' has overflowed its stack
PS C:\Users\user\Documents\git\test>
```

**Comment by Neohertz** [comment]
I'm experiencing the same issue.

**Comment by JohnnyMorganz** [comment]
I've released v1.3.2 which updates full-moon. Can you check if you still experience stack overflows now?

**Comment by ddavness** [comment]
Not on this particular package anymore! :)
